spec/lang/step/intrinsics.md

```diff
@@ -231,7 +231,7 @@ impl<M: Memory> Machine<M> {
 
         let alloc = self.mem.allocate(AllocationKind::Heap, size, align)?;
 
-        ret(Value::Ptr(alloc))
+        ret(Value::Ptr(alloc.widen(None)))
```

Comment:
Same here, needs a check above that the pointee is sized.

Reply:
So I understand it correctly, the ways to allocate a slice is to call the allocate intrinsic with the needed size, but return some raw pointer and then manually (unsafe) cast it to a wide pointer, e.g. with `from_raw_parts`? Which also means supporting the intrinsic `core::intrinsics::aggregate_raw_ptr` is essential? I mean the alternative is to have the alloc & dealloc macros be able to deal with slices themselves.

Reply:
The usual way to create a slice is something like
```rust
let x = [0; 32];
let y: &[i32] = &x;
```
In other words, it is created via an unsizing coercion.

`from_raw_parts` can be left to a later PR.

Reply:
Yes.

It will work for `Arc<[i32; N]>` to `Arc<[i32]>` though, that is also a legal unsizing coercion.